### PR TITLE
80X electron and photon energy regression for user analysis

### PR DESCRIPTION
**Pull request by rafaellopesdesa**
This is the current electron and photon energy and resolution regression recommended by the EGM group for 80X analysis. The documentation about it can be found here:

https://twiki.cern.ch/twiki/bin/view/CMS/EGMRegression (data-MC residuals will be release this week, but do not alter this PR).

This is a user-application and does not change any of the comparisons performed during tests. The reason why this does not go into RecoEgamma is because the structure of the regression changed substantially and the codes are no longer compatible (ie, it's not a simple change of configuration... even though the code is very similar... there was really no reason to do a different code all together). Since 80X cannot have changes in the RECO sequence, we would like to provide these energy regressions as a user-app code, as in this PR. This will only exist in 80X. In 81X and 90X the code will go into the RECO sequence (in RecoEgamma). The PRs will follow the inclusions of the tags in the GT, so that validation can be done. I've already requested the inclusion in the 90X GT and the PRs should follow shortly.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test
IIRC this PR doesn't lead to anything running in unit tests (no need to wait for the GT update to run compilation tests)

**Comment by monttj** [comment]
+1
This is the official recipe already being used in EGamma POG. 
It does not affect the official miniAOD workflow as described. 
It looks ok from my side.

File: EgammaAnalysis/ElectronTools/plugins/EGExtraInfoModifierFromDBUser.cc [hunk 1]
```diff
@@ -0,0 +1,741 @@
...
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+
+#include "RecoEgamma/EgammaTools/interface/EcalClusterLocal.h"
+#include "RecoEcal/EgammaCoreTools/interface/EcalClusterTools.h"
+#include "Geometry/CaloTopology/interface/CaloTopology.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "Geometry/CaloEventSetup/interface/CaloTopologyRecord.h"
+
+#include <vdt/vdtMath.h>
+#include <TFile.h>
```
Comment (davidlange6) [inline EgammaAnalysis/ElectronTools/plugins/EGExtraInfoModifierFromDBUser.cc]: Hi @rafaellopesdesa - this include is not needed. Please remove it

**Comment by davidlange6** [comment]
please test

**Pull request merged**